integer/src/lib.rs

```diff
@@ -665,6 +665,36 @@ impl_integer_for_usize!(u32, test_integer_u32);
 impl_integer_for_usize!(u64, test_integer_u64);
 impl_integer_for_usize!(usize, test_integer_usize);
 
+/// Calculate r * a / b, avoiding overflows and fractions.
+fn multiply_and_divide<T: Integer + Clone>(r: T, a: T, b: T) -> T {
+    // See http://blog.plover.com/math/choose-2.html for the idea.
+    let g = gcd(r.clone(), b.clone());
+    (r/g.clone()) * (a / (b/g))
```

Comment:
It's worth noting that this depends on the fact that `b` must evenly divide `r * a`, as that's what lets you know that `b/gcd(r,b)` divides `a` evenly.  Otherwise, you'd have to stick with the blog's original association `((r/g) · a)/(b/g)` to handle the truncation properly.